qa/tasks/ec_parity_consistency.py

```diff
@@ -0,0 +1,483 @@
+"""
+Use this task to check that parity shards in an EC pool
+match the output produced by the Ceph Erasure Code Tool.
+"""
+
+import logging
+import re
+import json
+import os
+import atexit
+import shutil
+from io import StringIO
+from io import BytesIO
+from tasks import ceph_manager
+from teuthology import misc as teuthology
+
+log = logging.getLogger(__name__)
+
+ERASURE_CODE_DIR = '/tmp/erasure-code/'
+DATA_SHARD_FILENAME = 'ec-obj'
+
+
+class ErasureCodeObject:
+    """
+    Store data relating to an RBD erasure code object,
+    including the object's erasure code profile as well as
+    the data for k + m shards.
+    """
+    def __init__(self, object_id: str, ec_profile: dict, object_size: int):
```

Comment:
Nice to see type signatures consistently for the methods and functions, supported since Python 3.5, I noticed the containers we build off Centos9 have Python 3.9 👍